godot-core/src/builtin/variant/mod.rs

```diff
@@ -231,6 +232,14 @@ impl Variant {
         unsafe { interface_fn!(variant_booleanize)(self.var_sys()) != 0 }
     }
 
+    /// Assuming that this is of type `OBJECT`, checks whether the object is dead.
+    ///
+    /// Does not check again that the variant has type `OBJECT`.
+    pub(crate) fn is_object_dead(&self) -> bool {
+        // There is no dedicated API, however the string repr has a special value for dead objects.
+        self.stringify() == "<Freed Object>".into()
```

Comment:
what about [is_instance_valid](https://github.com/godotengine/godot/blob/1586c5674bbafe1d7f4bd457d446f132b475890c/core/variant/variant_utility.cpp#L1218)? 

Note: [It is deprecated for public use](https://github.com/godot-rust/gdext/pull/901), but reserving it for internal use (`pub(crate)` won't allow to use it even in itest afaik) seems to be better idea than relying on repr for nullptr checks :sweat_smile:

Reply:
Oh, great idea! Totally forgot that that one takes variant as arguments :sweat_smile: who knew that would come in handy one day...